File: app/src/main/java/com/hover/stax/languages/Lang.java
package com.hover.stax.languages;

import android.content.Context;

import com.hover.stax.R;
import com.yariksoffice.lingver.Lingver;

import org.jetbrains.annotations.NotNull;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Locale;

public class Lang {
    public String code, name;

    public Lang(String code) {
        this.code = code;
        Locale l = new Locale(code);
        name = l.getDisplayLanguage(l);
        name = name.substring(0, 1).toUpperCase() + name.substring(1);
    }

    public boolean isSelected() {
        return code.equals(Lingver.getInstance().getLanguage());
    }

    @NotNull
    @Override
    public String toString() {
        return name;
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof Lang)) return false;
        Lang l = (Lang) other;
        return code.equals(l.code);
    }

    public static void LogChange(String code, Context c) {
        JSONObject data = new JSONObject();
        try {
            data.put("language", code);
        } catch (JSONException ignored) {
        }
        Utils.logAnalyticsEvent(c.getString(R.string.selected_language), data, c);
    }
}

File: app/src/main/java/com/hover/stax/languages/Lang.kt
/*
 * Copyright 2022 Stax
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.hover.stax.languages

import android.content.Context
import com.hover.stax.R
import com.hover.stax.core.AnalyticsUtil
import com.yariksoffice.lingver.Lingver
import java.util.*
import org.json.JSONException
import org.json.JSONObject
import timber.log.Timber

class Lang(val code: String) {

    var name: String? = null

    init {
        val locale = Locale(code)
        name = locale.getDisplayLanguage(locale).apply {
            substring(0, 1).uppercase().plus(substring(1))
        }
    }

    fun isSelected(): Boolean = code == Lingver.getInstance().getLanguage()

    override fun toString(): String = name!!

    override fun equals(other: Any?): Boolean {
        if (other !is Lang) return false

        return code == other.code
    }

    companion object {
        fun logChange(code: String, c: Context) {
            val data = JSONObject()
            try {
                data.put("language", code)
            } catch (e: JSONException) {
                Timber.e(e.localizedMessage)
            }

            com.hover.stax.utils.AnalyticsUtil.logAnalyticsEvent(c.getString(R.string.selected_language), data, c)
        }
    }
}